### PR TITLE
ethdb: Add paused write stat

### PR DESCRIPTION
**Pull request by rjl493456442**
Note: we need to wait this [pr](https://github.com/syndtr/goleveldb/pull/193) in goleveldb get merged and do the next :)

Implements #15347

### CODE REVIEW

**Comment by rjl493456442** [comment]
@karalabe PTAL

**Comment by karalabe** [comment]
Sweet :)

A few suggestions:

 * To truly make these measurements useful, it would be nice to actually print a warning message to the user whenever the delays exceed some threshold. Currently we just collect the measurements, but don't really act on it. A possible solution would be to define some tresholds and emit a warning if a collected stat is above that.
 * The stats gathering is only done if Geth is run with `--metrics`. That's fine as we don't want to gather all kinds of metrics all the time. However, if we want to warn the user about write delays, the write delay checking (not aggregation) should still run every now and again. A possible solution is to always run the `meter()` function, but skip most of it internally if we're not actually in metrics mode, only reacting to select tresholg exceeding events.

**Comment by rjl493456442** [comment]
Updated.

I run go-ethereum with fast sync mode at a clear state and dump write delay information to a file.
With the help of  [analysis script](https://gist.github.com/rjl493456442/03d642f17b3f0fe2de8ad28c2a5f1ce2), i find the 90% write delay number is about 214.75933 and 90% write delay duration is about 350 microseconds. 

So i set the write delay number threshold at 300, and write delay duration threshold at 500 microseconds. If the stats beyond the threshold, a warning log will been printed at user's console.

File: ethdb/database.go [hunk 1]
```diff
@@ -30,24 +30,30 @@ import (
...
 	"github.com/syndtr/goleveldb/leveldb/opt"
 
+	"fmt"
 	gometrics "github.com/rcrowley/go-metrics"
 )
 
-var OpenFileLimit = 64
+const (
+	writeDelayNThreshold = 250
+	writeDelayThreshold  = 500 * 1000 * 1000
```
Comment (karalabe) [inline ethdb/database.go]: Please use `500 * time.Millisecond` here.

**Comment by rjl493456442** [comment]
Superseded in #16499

**Pull request closed without merging**